feat: Add "Use Clerk with Next.js 12 and older" page (#543)
clerk/clerk-docs: docs/references/nextjs/usage-with-older-versions.mdx

---
title: Use Clerk with Next.js 12 and older
description: Learn how to use Clerk with older versions of Next.js.
---

# Use Clerk with Next.js 12 and older

> Clerk's [prebuilt components](/docs/components/overview) which are exported from the `@clerk/nextjs` package are leveraging APIs from Next.js itself. These APIs often change between major Next.js releases. While Clerk tries to offer the highest compatibility for any supported Next.js version, for Next.js 12 and older you'll need to modify your setup.

Reviewer: ```suggestion
Clerk's [prebuilt components](/docs/components/overview) are exported from the `@clerk/nextjs` package and leverage APIs from Next.js itself. These APIs often change between major Next.js releases. While Clerk tries to offer the highest compatibility for any supported Next.js version, for Next.js 12 and older, you need to modify your setup.
```

<Callout type="warning">
  Clerk highly recommends updating your Next.js version as older versions won't receive any updates in the future. Read [their upgrade guides](https://nextjs.org/docs/pages/building-your-application/upgrading) to learn more.
</Callout>

<Steps>

### Install `@clerk/nextjs` v4

> You need to install `^4.0.0` of `@clerk/nextjs`. Newer major versions only support Next.js 13+.

Reviewer: Using an active voice here. Instead of "you need to", we can simply go straight to telling the user what to do with "install".

```suggestion
Install `^4.0.0` of `@clerk/nextjs`. Newer major versions of `@clerk/nextjs` only support Next.js 13+.
```

<CodeBlockTabs type="installer" options={["npm", "yarn", "pnpm"]}>
  ```bash filename="terminal"
  npm install @clerk/nextjs@^4.0.0
  ```

  ```bash filename="terminal"
  yarn add @clerk/nextjs@^4.0.0
  ```

  ```bash filename="terminal"
  pnpm add @clerk/nextjs@^4.0.0
  ```
</CodeBlockTabs>

### Change your `next.config.js`

> As mentioned previously, the `@clerk/nextjs` components contain code for multiple Next.js versions. But depending on your version only certain components will be used. You need to instruct webpack to ignore modules on code paths that won't be used:

Reviewer: ```suggestion
As mentioned previously, the `@clerk/nextjs` components contain code for multiple Next.js versions, but depending on your version, only certain components will be used. Update your `next.config.js` to instruct webpack to ignore modules on code paths that won't be used.
```
